api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -2008,8 +2008,116 @@ type AzurePlatformSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SecurityGroupID is immutable"
 	// +kubebuilder:validation:Required
 	// +immutable
-	// +required
-	SecurityGroupID string `json:"securityGroupID,omitempty"`
+	SecurityGroupID string `json:"securityGroupID"`
+
+	// managedIdentities contains the managed identities needed for HCP control plane and data plane components that
+	// authenticate with Azure's API.
+	//
+	// +kubebuilder:validation:Required
+	ManagedIdentities AzureResourceManagedIdentities `json:"managedIdentities"`
+
+	// managementKeyVault contains information on the management cluster's Azure Key Vault. This Key Vault is where the
+	// managed identities certificates are stored. These certificates are pulled out of the Key Vault by the Secrets
+	// Store CSI driver and mounted into a volume on control plane pods requiring authentication with Azure API.
+	//
+	// More information on how the Secrets Store CSI driver works to do this can be found here:
+	// https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-driver.
+	//
+	// +kubebuilder:validation:Required
+	ManagementKeyVault ManagedAzureKeyVault `json:"managementKeyVault"`
+}
+
+// ManagedAzureKeyVault is the management cluster's Azure Key Vault where the managed identity certificates are stored.
+type ManagedAzureKeyVault struct {
```

Comment:
Isn't this Key Vault called "Managemenet Key Vault" instead of "Managed Key Vault" ?

cc @geoberle 

Asking to try to be consistent with the naming in all components

Reply:
Is there some other component where its named "Management Key Vault"?

Reply:
I kept this the same, but renamed the one for the managed identities in the latest push